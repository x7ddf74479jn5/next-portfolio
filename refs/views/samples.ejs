<%- include ('components/_head') %>
<%- include ('components/_header') %>



<% const samples =
[
  {
    title: "ポートフォリオ",
    category: "Web制作｜デザイン｜レスポンシブ｜アニメーション",
    href: "/samples/portfolio/",
    img: {
      src: "/img/samples/portfolio-eyecatch.png",
      alt: "Thumbnail of portfolio"
    },
  },
  {
    title: "ECサイト",
    category: "Web制作｜デザイン｜レスポンシブ|SPA",
    href: "/samples/ec/",
    img: {
      src: "/img/samples/ec-app.jpeg",
      alt: "Thumbnail of EC site"
    },
  },
  {
    title: "チャットボット",
    category: "Webアプリ開発｜チャット｜SPA",
    href: "/samples/chat/",
    img: {
      src: "/img/samples/chat.jpeg",
      alt: "Thumbnail of Chat Bot"
    },
  },

]
%>





<%# content %>
<main>
  <section>
    <div class="module-spacer--medium"></div>
  </section>

  <section class="c-section">
    <div class="c-section__area-title animated fadeInLeft">
      <div class="p-headline__sub animated fadeInUp fast delay-1s">
        <h2>Samples</h2>
        <p>制作事例</p>
        <ul class="u-text__breadcrumbs">
          <li><a href="/" target="_self">HOME</a><span>＞</span></li>
          <li><a href="/samples/" target="_self">SAMPLES</a><span></span></li>
        </ul>
      </div>
    </div>
  </section>

  <section>
    <div class="module-spacer--medium"></div>
  </section>

  <section class="c-section">
    <div class="c-section-wrapin-wide">
      <ul class="p-list__works">
        <% for (const sample of samples) { %>
          <li>
            <a class="u-text__line-none" href="<%= sample.href %>" >
              <p class="p-media__thumb">
                <img src="<%= sample.img.src %>" alt="<%= sample.img.alt %>" >
              </p>
              <div class="p-list__works-detail">
                <h3 class="title"><%= sample.title %></h3>
                <p class="category"><%= sample.category %></p>
                <span class="icon"></span>
              </div>
            </a>
          </li>
          <% } %>
      </ul>
    </div>
  </section>
</main>
<%# /content %>

<%- include ('components/_footer') %>