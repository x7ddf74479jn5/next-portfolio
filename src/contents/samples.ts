import { identity } from "../utils/helper";
import { images } from "../utils/images";
import { externalLinks, internalLinks } from "../utils/paths";

export type Sample = {
  id: string;
  imgSrc: string;
  imgAlt: string;
  category: string;
  link: string;
  title: string;
  description: string;
};

export const samples: Sample[] = [
  identity({
    id: "portfolio",
    title: "ポートフォリオ",
    category: "Web制作｜デザイン｜レスポンシブ｜アニメーション",
    description: "現在閲覧してくださっているサイトです。",
    link: internalLinks.portfolio,
    imgSrc: images.portfolio.path,
    imgAlt: images.portfolio.alt,
  }),
  identity({
    id: "chat-bot",
    title: "チャットボット",
    category: "Webアプリ開発｜チャット｜SPA",
    description: "事前に用意したQAを使って自動応答するチャットボットです。",
    link: internalLinks.chat,
    imgSrc: images.chat.path,
    imgAlt: images.chat.alt,
  }),
  identity({
    id: "ec-app",
    title: "ECサイト(開発中)",
    category: "Webアプリ開発｜EC｜SPA",
    description: "React学習用に開発しているデモサイトです。",
    link: internalLinks.ec,
    imgSrc: images.ec.path,
    imgAlt: images.ec.alt,
  }),
];

export type SampleDetail = {
  id: string;
  img: string;
  alt: string;
  summary: string;
  purposes: string[];
  points: string[];
  langs: string;
  techs: string;
  link: {
    href?: string;
    label?: string;
    description?: string;
  };
  note?: string;
};

export const sampleDetails: SampleDetail[] = [
  identity({
    id: "portfolio",
    img: images.portfolio.path,
    alt: images.portfolio.alt,
    summary: "ポートフォリオサイトです。",
    purposes: ["Next.js/TypeScriptの学習", "転職活動のため"],
    points: [
      "Webサイト制作に必要不可欠な要素は抑えた",
      "CSSはなるべくシンプルに",
      "JavaScriptの使用も最小限に",
      "見ていて飽きさせないためのCSSアニメーション",
      "SEOを意識したHTML構造",
      "StorybookやCIを使ったモダンな開発環境",
    ],
    langs: "React, TypeScript, Sass",
    techs: "Next.js, Vercel, CSS Modules, Storybook, GitHub Actions",
    link: {
      description: "このサイトなので省略。",
    },
    note: " 実験的な変更を含め、日々改良中です。もしデザイン崩れや機能不調が確認できましたら、お問い合わせページからご一報くださると助かります。",
  }),
  identity({
    id: "chat",
    img: images.chat.path,
    alt: images.chat.alt,
    summary: "ユーザーに質問を投げて自動で応答するチャットボット",
    purposes: ["カジュアルにWebサイトへのお問い合わせをしてもらうため"],
    points: ["SPAらしい画面描画の切り替わり", "問い合わせ結果をSlackへ通知", "チャット風のデザインとアニメーション"],
    langs: "React, Sass",
    techs: "Material-UI, Firebase Hosting, Firestore, Cloud Functions",
    link: {
      href: externalLinks.chat,
      label: "チャットボット",
    },
    note: "ヘッダーメニューのロボットアイコンからお試しいただけます。",
  }),
  identity({
    id: "ec",
    img: images.ec.path,
    alt: images.ec.alt,
    summary: "ECアプリ（開発中です）",
    purposes: ["React-ReduxでWebアプリケーションの作り方を学習するため"],
    points: ["SPAらしい画面描画の切り替わり", "ECアプリとしての機能を順次追加中"],
    langs: "React, Sass",
    techs: "Redux, Material-UI, Firebase Hosting, Firestore, Cloud Functions",
    link: {
      href: externalLinks.ec,
      label: "ECアプリ",
    },
    note: "ログイン時に入力してください。\nemail: psexample86@gmail.com\npassword: 8ox2lkmw",
  }),
];
