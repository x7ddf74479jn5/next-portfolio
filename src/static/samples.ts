type Sample = {
  img: {
    src: string;
    alt: string;
  };
  category: string;
  href: string;
  title: string;
  description: string;
};

import { images, links } from "./links";

export const samples: Sample[] = [
  {
    title: "ポートフォリオ",
    category: "Web制作｜デザイン｜レスポンシブ｜アニメーション",
    description: "現在閲覧してくださっているサイトです。",
    href: "/samples/portfolio/",
    img: {
      src: images.portfolio,
      alt: "Thumbnail of portfolio",
    },
  },
  {
    title: "チャットボット",
    category: "Webアプリ開発｜チャット｜SPA",
    description: "事前に用意したQAを使って自動応答するチャットボットです。",
    href: "/samples/chat/",
    img: {
      src: images.chat,
      alt: "Thumbnail of Chat Bot",
    },
  },
  {
    title: "ECサイト(開発中)",
    category: "Webアプリ開発｜EC｜SPA",
    description: "React学習用に開発しているデモサイトです。",
    href: "/samples/ec/",
    img: {
      src: images.ec,
      alt: "Thumbnail of EC site",
    },
  },
];

export type SampleDetail = {
  id: string;
  img: string;
  alt: string;
  abstract: string;
  purpose: string[];
  point: string[];
  lang: string;
  tech: string;
  link: { href: string; label: string } | string;
  title: string;
  caption: string;
};

export const samplesDetail: SampleDetail[] = [
  {
    id: "portfolio",
    img: images.portfolio,
    alt: "Eye-catch of portfolio",
    abstract: "ポートフォリオサイトです。",
    purpose: ["Next.js/Typescriptの学習", "転職活動のため", "案件獲得のため"],
    point: [
      "Webサイト制作に必要不可欠な要素は抑えた",
      "CSSはなるべくシンプルに",
      "JavaScriptの使用も最小限に",
      "見ていて飽きさせないためのCSSアニメーション",
      "SEOを意識したHTML構造",
    ],
    lang: "HTML5, CSS3, JavaScript(ES6) Typescript ",
    tech: "React Next.js Vercel Sass",
    link: "このサイトなので省略",
    title: "Portfolio",
    caption: "",
  },
  {
    id: "chat-bot",
    img: images.chat,
    alt: "Eye-catch of chat-bot",
    abstract: "ユーザーに質問を投げて自動で応答するチャットボット",
    purpose: ["カジュアルにWebサイトへのお問い合わせをしてもらうため"],
    point: ["SPAらしい画面描画の切り替わり", "問い合わせ結果をSlackへ通知", "チャット風のデザインとアニメーション"],
    lang: "JavaScript(React), JSX",
    tech: "Sass, React, Material-ui, Firebase Hosting, Firestore, Cloud Functions",
    link: { href: "chatbot-demo-70752.web.app", label: "チャットボット" },
    title: "Chat App",
    caption: "",
  },
  {
    id: "EC-app",
    img: images.ec,
    alt: "Eye-catch of EC-app",
    abstract: "ECアプリ（開発中です）",
    purpose: ["React-ReduxでWebアプリケーションの作り方を学習するため"],
    point: ["SPAらしい画面描画の切り替わり", "ECアプリとしての機能を順次追加中"],
    lang: "JavaScript(React), JSX",
    tech: "Sass, React, Redux, Material-ui, Firebase Hosting, Firestore, Cloud Functions",
    link: { href: "https://ec-app-a2f5e.web.app/", label: "ECアプリ" },
    title: "Chat App",
    caption: "",
  },
];
