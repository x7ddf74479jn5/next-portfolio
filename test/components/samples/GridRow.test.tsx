import { render } from "@testing-library/react";
import { GridRowLink, GridRowList, GridRowParagraph } from "src/components/samples/GridRow";

describe("GridRow", () => {
  describe("GridRowLink", () => {
    it("matches snapshot", () => {
      const { asFragment } = render(
        <GridRowLink header="header" data={{ href: "href", label: "label", description: "description" }} />
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      const { getByText, container, getByRole } = render(
        <GridRowLink header="header" data={{ href: "href", label: "label", description: "description" }} />
      );
      const headerElement = getByRole("heading", { level: 3 });
      expect(headerElement).toHaveTextContent("header");
      const descriptionElement = getByText("description");
      expect(descriptionElement).toBeInTheDocument;

      const svgElement = container.getElementsByTagName("svg");
      expect(svgElement).toBeInTheDocument;
    });
  });

  describe("GridRowList", () => {
    it("matches snapshot", () => {
      const { asFragment } = render(<GridRowList header="header" data={["data-1", "data-2", "data-3"]} />);
      expect(asFragment()).toMatchSnapshot();
    });
    it("rendered correctly", () => {
      const { getAllByRole, getByRole } = render(<GridRowList header="header" data={["data-1", "data-2", "data-3"]} />);
      const headerElement = getByRole("heading", { level: 3 });
      expect(headerElement).toHaveTextContent("header");
      const listElements = getAllByRole("listitem");
      expect(listElements).toHaveLength(3);
      expect(listElements[0]).toHaveTextContent(/data/);
    });
  });

  describe("GridRowParagraph", () => {
    it("matches snapshot", () => {
      const { asFragment } = render(<GridRowParagraph header="header" data="data" />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  it("rendered correctly", () => {
    const { container, getByRole } = render(<GridRowParagraph header="header" data="data" />);
    const headerElement = getByRole("heading", { level: 3 });
    expect(headerElement).toHaveTextContent("header");
    expect(container).toHaveTextContent("data");
  });
});
