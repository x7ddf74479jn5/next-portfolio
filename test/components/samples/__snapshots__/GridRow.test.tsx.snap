// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridRow GridRowLink matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      header
    </h3>
    <p>
      description
    </p>
    <p>
      <a
        href="href"
        rel="noreferrer"
        target="_blank"
      >
        <span
          class="externalLink"
        >
          label
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </span>
      </a>
    </p>
  </div>
</DocumentFragment>
`;

exports[`GridRow GridRowList matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      header
    </h3>
    <ul>
      <li>
        data-1
      </li>
      <li>
        data-2
      </li>
      <li>
        data-3
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`GridRow GridRowParagraph matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      header
    </h3>
    <p>
      data
    </p>
  </div>
</DocumentFragment>
`;
